doc/source/whatsnew/v0.25.1.rst

```diff
@@ -21,6 +21,8 @@ Other enhancements
 Bug fixes
 ~~~~~~~~~
 
+- Bug in :func:`fill_na` in :meth:`Categorical.fillna` would replace all values, not just those that are NaN (:issue:`26215`)
```

Comment:
- Can you move this to the Categorical section a couple lines below?
- You can remove the <code>in :func:\`fill_na\`</code> since you added the more specific `Categorical.fillna`
- Can you add double backticks around NaN (`NaN` --> <code>\`\`NaN\`\`</code>)

Reply:
Sure, thanks for your review!